src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -131,10 +131,13 @@ void CoreWorkerDirectTaskSubmitter::AddWorkerLeaseClient(
     const google::protobuf::RepeatedPtrField<rpc::ResourceMapEntry> &assigned_resources,
     const SchedulingKey &scheduling_key,
     const TaskID &task_id) {
-  client_cache_->GetOrConnect(addr.ToProto());
   int64_t expiration = current_time_ms() + lease_timeout_ms_;
-  LeaseEntry new_lease_entry = LeaseEntry(
-      std::move(lease_client), expiration, assigned_resources, scheduling_key, task_id);
+  LeaseEntry new_lease_entry{std::move(lease_client),
```

Comment:
@ericl it's stored in the lease entry which means, there are always a shared_ptr when the worker is in use.

In the new way, the user needs to keep the client and manage the lifecycle, which I think it's reasonable since the caller knows the lifecycle of a core worker. And we actually maintain this in our code.

The task transport is always going to get client from lease entry. Same for GCS side, where client should always be fetched from GcsLeasedWorker.


The client pool is for different component to share the connections if there is one.